src/QuadVelocityController.cpp

```diff
@@ -31,27 +33,33 @@ void QuadVelocityController::setTargetVelocity(geometry_msgs::Twist twist)
 {
     throttle_pid_.setSetpoint(twist.linear.z);
     pitch_pid_.setSetpoint(twist.linear.x);
-    roll_pid_.setSetpoint(twist.linear.y);
+    roll_pid_.setSetpoint(-twist.linear.y);
     yaw_pid_.setSetpoint(twist.angular.z);
 }
 
 // Needs to be called at regular intervals in order to keep catching the latest velocities.
-iarc7_msgs::OrientationThrottleStamped QuadVelocityController::update()
+iarc7_msgs::OrientationThrottleStamped QuadVelocityController::update(const ros::Time& time)
 {
-    // Get the time delta
     geometry_msgs::Vector3 velocities;
     getVelocities(velocities);
-    double time_delta = 0.0;
+
+    // Get the time delta
+    if (last_time_ == ros::Time(0)) {
+        last_time_ = time;
```

Comment:
A delta of 0 is a valid input to the PID.

The PID is always initialized upon object creation this was by design.

I would argue that warnings should always be sent when receiving a time delta of zero in a PID controller.

However because this class is a quad controller and there is no reason it shouldn't wait to be fully initialized before sending flight commands.